Toolkit/Public/Get-RscAzureNativeVm.ps1

```diff
@@ -0,0 +1,95 @@
+#Requires -Version 3
+function Get-RscAzureNativeVm {
+    <#
+    .SYNOPSIS
+    Retrieves RscAzureNativeVm objects protected by Rubrik Security Cloud
+
+    .DESCRIPTION
+    This cmdlet uses the GQL query 'azureNativeVirtualMachines' to retrieve a list of VMs with a predetermined set of properties.
+
+    .LINK
+    Schema reference:
+    https://rubrikinc.github.io/rubrik-api-documentation/schema/reference
+
+    .EXAMPLE
+    # Get all
+    Get-RscAzureNativeVm
+
+    .EXAMPLE
+    # Get object with specific name
+    Get-RscAzureNativeVm -Name "jake-001"
+
+    .EXAMPLE
+    # Get objects by specifying part of a name
+    Get-RscAzureNativeVm -Name "*jake*"
+    #>
+
+    [CmdletBinding(
+        DefaultParameterSetName = "Name"
+    )]
+    Param(
+        [Parameter(
+            Mandatory = $false,
+            ParameterSetName = "Id"
+        )]
+        [String]$Id,
+        [Parameter(
+            Mandatory = $false,
+            ParameterSetName = "Name"
+        )]
+        [String]$Name,
+        [Parameter(
+            Mandatory = $false,
+            ValueFromPipeline = $true,
+            ParameterSetName = "Name"
+        )]
+        [RubrikSecurityCloud.Types.GlobalSlaReply]$Sla
+    )
+    
+    Process {
+
+       # The query is different for getting a single object by ID.
+        if ($Id) {
+            $query = New-RscQuery -GqlQuery azureNativeVirtualMachine
+            $query.var.filter = @()
+            $query.Var.fid = $Id
+
+            $result = Invoke-Rsc -Query $query
+            $result
+        } else {
+            $query = New-RscQuery -GqlQuery azureNativeVirtualMachines
+            $query.var.filter = @()
+
+            if ($Name) {
+                $nameFilter = New-Object -TypeName RubrikSecurityCloud.Types.Filter
+                # Regex filter doesn't work in the API right now, but we're going to play pretend. 
+                # With real Regex, users could search for VMs that start with the letter A if they wanted.
+                if ($name.Contains("*")) {
+                    $name.Replace("*",'')
```

Comment:
Replace() doesn't replace in place - I'd take out this line, since you call $name.Replace below

Reply:
Yeah, actually the name filtering needs to be completely rewritten for this one. It doesn't follow the filtering pattern for CDM workloads.